src/tartiflette_asgi/_endpoints.py

```diff
@@ -33,16 +33,36 @@ async def get(self, request: Request) -> Response:
 
 class GraphQLEndpoint(HTTPEndpoint):
     async def get(self, request: Request) -> Response:
-        return await self._get_response(request, data=request.query_params)
+        variables = None
+        if "variables" in request.query_params:
+            try:
+                variables = json.loads(request.query_params["variables"])
+            except json.JSONDecodeError:
+                return JSONResponse(
+                    {"error": "Unable to decode variables: Invalid JSON."}, 400
+                )
+        return await self._get_response(
+            request, data=request.query_params, variables=variables
+        )
 
     async def post(self, request: Request) -> Response:
         content_type = request.headers.get("Content-Type", "")
 
+        variables = None
+        if "variables" in request.query_params:
+            try:
+                variables = json.loads(request.query_params["variables"])
```

Comment:
I am not an expert of graphQL. But what would happens if the variables are empty? Is that possible? L65 will be empty and we will pass also empty variabes in L74. Is the L57 of tests/test_graphql_api.py testing empty variables? I wonder how the answer is "hello stranger" and if this will work when using it by a third party.

Reply:
There are two cases here:
1. Not setting any variables at all. This is the default behaviour - variable handling is null-aware, so queries that don't require variables and don't receive them at all will succeed. This is the case for the test you mentioned.
2. Setting the `variables` parameter to an empty string. Unfortunately this is not covered by the GraphQL spec, so a GraphQL server could treat it either as malformed JSON (because it doesn't parse) or as unset (because it's empty). In its current state, `tartiflette-asgi` will throw a malformed JSON error, which is the same behaviour as for Apollo Server (faas-middleware) and the Django GraphQL layer (herbarium).

I think we're good here - we're not breaking any existing clients of the upstream library because passing GET variables didn't work at all before, and ensuring well-formed arguments is the client's responsibility as well.